xtext/org.icyphy.linguafranca/src/org/icyphy/generator/ReactionInstance.xtend

```diff
@@ -100,19 +101,20 @@ class ReactionInstance extends NamedInstance<Reaction> {
                 var portInstance = parent.lookupPortInstance(effect)
                 if (portInstance instanceof MultiportInstance) {
                     for (multiportInstance : portInstance.instances) {
-                        this.dependentPorts.add(multiportInstance)
+                        this.effects.add(multiportInstance)
                         multiportInstance.dependsOnReactions.add(this)
                     }
                 } else {
-                    this.dependentPorts.add(portInstance)
+                    this.effects.add(portInstance)
                     portInstance.dependsOnReactions.add(this)
                 }
             } else {
+                // FIXME: Unused.
```

Comment:
This commit seems to comment out construction of some graph information that we might need in the future (dependencies across actions, for example). I guess this is OK, but is there a reason for this? Does it create a significant performance hit?  It will probably be needed again when we calculate safe-to-process offsets.

Reply:
The concern was not performance. There were a lot of `dependsOn` and `dependentOn` variables and I had a hard time remembering what they were being used for. It turned out that these ones weren't used at all. I can reinstate them, but I think it would be better to lump them in with `sources` and `effects` (as they are also in the language) and use filters to separate ports from actions and timers. If you agree, I'll carry out this refactoring before the merge.

Reply:
OK, sounds reasonable.

Reply:
This is done in https://github.com/icyphy/lingua-franca/commit/15ff7abc03d195998b6fcdfbcb6c4e74f87ee0ac.